src/create-matcher.js

```diff
@@ -198,6 +203,26 @@ function fillParams (
   }
 }
 
+function getParams (path: string): Array<string> {
+  const hit = regexpParamsCache[path]
+
+  if (hit) {
+    return hit
+  }
+
+  const keys = []
+  const paramNames: Array<string> = []
+
+  Regexp(path, keys)
+
+  for (let i = 0, len = keys.length; i < len; ++i) {
+    const key = keys[i]
+    if (key && 'name' in key) paramNames.push(key.name)
+  }
```

Comment:
This whole for-loop can be simplified to `keys.map(k => k.name)` for brevity. Since we are already using cache here, the performance penalty of using a `Array.prototype.map` is negligible.